### PR TITLE
Avoid specializing when creating Observable{Any}

### PR DESCRIPTION
**Pull request by timholy**
Discovered while analyzing Makie's TTFP.

### CODE REVIEW

**Comment by timholy** [comment]
Note that this commit points out a problem with `convert` methods; someone who knows more about their history should fix. `blame` flags 6ff7370816c2ecd0245ff18a81a3d47ca7269653, cc @SimonDanisch.

File: src/Observables.jl [hunk 1]
```diff
@@ -38,6 +40,9 @@ Observable(val::T) where {T} = Observable{T}(val)
 
 observe(x::Observable) = x
 
+# These methods seem backwards: Observable should be swapped with AbstractObservable, because when I ask
+# to convert to a concrete type I expect to get that type.
+# They are also the source of an ambiguity.
```
Comment (SimonDanisch) [inline src/Observables.jl:45]: Makes sense! I'm not even sure if these are used :D

Comment (timholy) [reply src/Observables.jl:45]: As is this PR breaks AbstractPlotting at the build step, but if you delete them then everything builds. I haven't tested any farther than that.

**Comment by SimonDanisch** [comment]
Oh, didn't realize I'm pushing directly to this branch... Well, this seems to pass & I also let the whole Makie testsuite run over it and there seem to be no regressions :) 
@shashi  good to merge?

**Comment by SimonDanisch** [comment]
Considering that Makie doesn't break, which should test most corner cases, I'll merge this ;)
